src/ess/nmx/mcstas_loader.py

```diff
@@ -0,0 +1,81 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+from typing import Iterable, NewType, Optional
+
+import scipp as sc
+import scippnexus as snx
+
+PixelIDs = NewType("PixelIDs", sc.Variable)
+InputFilepath = NewType("InputFilepath", str)
+NMXData = NewType("NMXData", sc.DataArray)
+
+# McStas Configurations
+MaximumProbability = NewType("MaximumProbability", int)
+DefaultMaximumProbability = MaximumProbability(100_000)
+
+
+def _retrieve_event_list_name(keys: Iterable[str]) -> str:
+    prefix = "bank01_events_dat_list"
+
+    # (weight, x, y, n, pixel id, time of arrival)
+    mandatory_fields = 'p_x_y_n_id_t'
+
+    for key in keys:
+        if key.startswith(prefix) and mandatory_fields in key:
+            return key
+
+    raise ValueError("Can not find event list name.")
+
+
+def _copy_partial_var(
+    var: sc.Variable, idx: int, unit: Optional[str] = None, dtype: Optional[str] = None
+) -> sc.Variable:
+    """Retrieve a property from a variable."""
+    var = var['dim_1', idx].astype(dtype or var.dtype, copy=True)
+    if unit:
+        var.unit = sc.Unit(unit)
+    return var
+
+
+def _get_mcstas_pixel_ids() -> PixelIDs:
+    """pixel IDs for each detector"""
+    intervals = [(1, 1638401), (2000001, 3638401), (4000001, 5638401)]
+    ids = [sc.arange('id', start, stop, unit=None) for start, stop in intervals]
+    return PixelIDs(sc.concat(ids, 'id'))
+
+
+def load_mcstas_nexus(
+    file_path: InputFilepath,
+    max_prop: Optional[MaximumProbability] = None,
```

Comment:
```suggestion
    max_prob: Optional[MaximumProbability] = None,
```
or do not abbreviate?